jfrog-cli/artifactory/commands/golang/go.go

```diff
@@ -114,12 +124,51 @@ func ExecuteGo(noRegistry bool, goArg, targetRepo, buildName, buildNumber string
 		if err != nil {
 			return err
 		}
+		err = goProject.LoadDependencies()
+		if err != nil {
+			return err
+		}
 		err = utils.SaveBuildInfo(buildName, buildNumber, goProject.BuildInfo(false))
 	}
 
 	return err
 }
 
+func retrieveAndPublish(targetRepo, goArg string, depsTidy bool, details *config.ArtifactoryDetails) error {
+	// Need to run Go without Artifactory to resolve all dependencies.
+	cache := goutils.GetStaticCache()
```

Comment:
The cache is stored as a static variable.
Can we use it as a local variable instead? It may be safer this way.

Reply:
Yes, changed the cache to be local variable.